tee-worker/litentry/core/assertion-build/src/a7.rs

```diff
@@ -21,64 +21,48 @@ compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the sam
 extern crate sgx_tstd as std;
 
 use crate::*;
-use itp_stf_primitives::types::ShardIdentifier;
-use itp_types::AccountId;
-use itp_utils::stringify::account_id_to_string;
-use lc_credentials::Credential;
 use lc_data_providers::{
 	achainable::{AchainableClient, AchainableQuery, VerifiedCredentialsIsHodlerIn},
 	vec_to_string,
 };
-use litentry_primitives::SupportedNetwork;
-use log::*;
-use std::{string::ToString, vec, vec::Vec};
 
 const VC_A7_SUBJECT_DESCRIPTION: &str =
 	"The user has been consistently holding at least {x} amount of tokens before 2023 Jan 1st 00:00:00 UTC on the supporting networks";
 const VC_A7_SUBJECT_TYPE: &str = "DOT Holding Assertion";
 const VC_A7_SUBJECT_TAG: [&str; 1] = ["Polkadot"];
 
-pub fn build(
-	identities: Vec<Identity>,
-	min_balance: ParameterString,
-	shard: &ShardIdentifier,
-	who: &AccountId,
-) -> Result<Credential> {
-	debug!(
-		"Assertion A7 build, who: {:?}, identities: {:?}",
-		account_id_to_string(&who),
-		identities
-	);
+pub fn build(req: &AssertionBuildRequest, min_balance: ParameterString) -> Result<Credential> {
+	debug!("Assertion A7 build, who: {:?}", account_id_to_string(&req.who),);
 
 	let q_min_balance = vec_to_string(min_balance.to_vec()).map_err(|_| {
 		Error::RequestVCFailed(Assertion::A7(min_balance.clone()), ErrorDetail::ParseError)
 	})?;
 
 	let mut client = AchainableClient::new();
-	let mut addresses = vec![];
+	let identities = transpose_identity(&req.vec_identity);
```

Comment:
I think we should filter out all "identities" having `Web3Network` other than Polkadot as it doesn't make sense to ask for them in context of this assertion or it may lead to false positive results.

Reply:
Maybe this is what you meant? https://github.com/litentry/litentry-parachain/blob/08a7b9201035753dc8d344b69bc09e80ac3c25be/tee-worker/app-libs/stf/src/trusted_call_litentry.rs#L141-L148

Reply:
Yes, but the purpose of this code is to reduce request size, not to prevent assertions from making wrong requests.
I think it could be worth to add redundant check. Better safe than sorry.

Reply:
I don't get it - the `assertion_networks` already contains the desired network, doesn't it? E.g. for `A7` it would be `Web3Network::Polkadot`

Reply:
Yes, the code in current state should work fine.